### PR TITLE
Create migration notes in Rust docs for v22

### LINKED ISSUE #1048: Create MIGRATION.md file
### What problem does your feature solve?
If there are breaking changes, users must spend sometimes a significant chunk of time to go through changelog and most often commits as well to see what has changed and what needs to be adjusted in code.

### What would you like to see?
Create a markdown file, where breaking changes would be described with examples before and after.

### What alternatives are there?
Having multiple users asking same questions on discord.

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add migration notes for v22.

### Screenshot
soroban_sdk___migrating - Rust

### Why
To make the most of docs.rs to provide more complete and thorough migration notes that have not only textual descriptions, but examples and working code that can be tested.

This change adds a GitHub Actions workflow that checks that for any new releases we are preparing, that the migration notes are added.

Close #1048

### CODE REVIEW

**Comment by sisuresh** [comment]
So in the release workflow, a PR like https://github.com/stellar/rs-soroban-sdk/pull/1244 will catch the missing notes, and then we'd add it in that PR?

**Comment by leighmcculloch** [comment]
Yup, that's correct. The job will error indicating that the release notes need writing.

**Comment by leighmcculloch** [comment]
I think it's also reasonable for us to go the simpler route of making the release notes part of the GitHub Release. If we want to include working examples in the release notes then the Rust docs will help us do that in a reliable way.

**Review by sisuresh — DISMISSED** [review]

**Review by dmkozh — DISMISSED** [review]

**Comment by leighmcculloch** [comment]
I missed getting this change in before the last release, and reflecting on this I'm not convinced we're going to be able to maintain keeping this updated and doing it before every release which is an annoying requirement because the release notes get snapshotted at the point in time the tag is made. Closing because while it seemed like a nice idea I don't expect we'll be able to execute it very well.

We should stick to GitHub release notes, and where-possible including working examples inside the Rust docs for any new feature or changing feature, along with using the rust deprecated annotations to help folks get to where they need to be.

**Comment by leighmcculloch** [comment]
Reopening because it occurred to me that why-not-both. We can do release notes on GitHub releases as we do for the simple release flow, and have a _migrations section of the rust docs dedicated to just breaking changes. 

This is more similar to what @ueco-jb suggested in https://github.com/stellar/rs-soroban-sdk/issues/1048.

File: soroban-sdk/src/_migrating.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
+//! # Migrating from v21 to v22
+//!
+//! 1. [`Env::register`] and [`Env::register_at`] replace [`Env::register_contract`] and [`Env::register_contract_wasm`].
+//!
+//!    [`register`] registers both native contracts previously registered with
+//!    [`register_contract`] and WASM contracts previously registered with
```
Comment (dmkozh) [inline soroban-sdk/src/_migrating.rs:6]: nit: WASM->Wasm

We weren't too consistent on this, but it seems like most of the docs should now be using the 'official' abbreviation (including most of the SDK doc comments as well).

**Review by dmkozh — APPROVED** [review]

File: soroban-sdk/src/_migrating.rs (hunk 1 above)
Comment (leighmcculloch) [reply soroban-sdk/src/_migrating.rs:6]: Ah yeah, you're right, it should be `Wasm`. I get tripped on this after typing `WASM` in constants so much.
```suggestion
//!    [`register_contract`] and Wasm contracts previously registered with
```

File: soroban-sdk/src/_migrating.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+//!    }
+//!
+//!    #[test]
+//!    fn test() {
+//!    # }
+//!    # #[cfg(feature = "testutils")]
+//!    # fn main() {
+//!        let env = Env::default();
+//!        let address = Address::generate(&env);
+//!        env.register(
```
[the remark this answers is not in the document]
Comment (leighmcculloch) [reply soroban-sdk/src/_migrating.rs:61]: Oh it is. How did this compile?!

[the remark this answers is not in the document]
Comment (leighmcculloch) [reply soroban-sdk/src/_migrating.rs]: Thanks @jayz22 for catching this. Turned out it wasn't testing any of the tests in here and they were horribly horribly broken. It took quite a few changes to get it fixed 😬.

**Pull request merged**